src/event-scope.ts

```diff
@@ -1,177 +1,191 @@
-import { Event, EventSubscriber } from "./events";
+import { Event, EventObjectImpl, EventSubscriber, EventSubscription } from "./events";
 import { getEventSubscriptions } from "./helpers";
 
-export let EventScope$current: EventScope = null;
+export interface EventScopeSettings {
+	/**
+	 * Controls the maximum number of times that a child event scope can transfer events to its parent
+	 * while the parent scope is exiting. A large number indicates that the consumer of the event scope
+	 * is likely cycling through the same activities repeatedly (i.e. infinite loop). This setting can
+	 * be used to cause the event scope to exit before the browser chooses to terminate the script in
+	 * a way that it may not be able to fully recover from.
+	 */
+	maxExitingTransferCount: number;
+
+	/**
+	 * Controls the maximum depth that an event scope can reach. A large number indicates that the consumer
+	 * of the event scope is likely cycling through the same activities repeatedly (i.e. infinite recursion)
+	 * and the browser's "max stack" limit will likely be reached at some point. This setting can be
+	 * used to cause the event scope to exit before the limit is reached, in order to avoid the browser
+	 * terminating the script in a way that it may not be able to fully recover from.
+	 */
+	maxDepth: number;
+}
 
-// TODO: Make `nonExitingScopeNestingCount` an editable configuration value
-// Controls the maximum number of times that a child event scope can transfer events
-// to its parent while the parent scope is exiting. A large number indicates that
-// rules are not reaching steady-state. Technically something other than rules could
-// cause this scenario, but in practice they are the primary use-case for event scope.
-const nonExitingScopeNestingCount = 100;
+export const EVENT_SCOPE_DEFAULT_SETTINGS: EventScopeSettings = {
+	maxExitingTransferCount: 100,
+	maxDepth: 1000
```

Comment:
This could be set to a smaller, sane default. However, one potential down-side is that it seems possible for a somewhat large number of scopes to be created if there are a lot of rules that depend on one another, and a change triggers the whole "chain" to execute.